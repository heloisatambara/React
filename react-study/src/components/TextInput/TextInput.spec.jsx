import { TextInput } from ".";

import { React, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("<TextInput/>", () => {
    it("should have a value of searchValue", () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} searchValue={"test"} />);

        const input = screen.getByPlaceholderText(/Search posts/i);
        expect(input.value).toBe("test");
    });

    it("should call handleChange function on each key pressed", () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} searchValue={"value"} />);

        const input = screen.getByPlaceholderText(/Search posts/i);

        const value = "value";

        userEvent.type(input, value);

        expect(input.value).toBe("value");
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it("should match snapshot", () => {
        const fn = jest.fn();
        const { container } = render(
            <TextInput handleChange={fn} searchValue={"test"} />
        );
        expect(container).toMatchSnapshot();
    });
});
