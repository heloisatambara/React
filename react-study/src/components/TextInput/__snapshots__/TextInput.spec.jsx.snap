// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Search posts"
    type="search"
    value=""
  />
</div>
`;
